src/juju/jimm/api.ts

```diff
@@ -24,51 +24,78 @@ export const endpoints = () => {
 /**
   Fetch audit events via the JIMM facade on the given controller connection.
  */
-export function findAuditEvents(
+export async function findAuditEvents(
   conn: ConnectionWithFacades,
   params?: FindAuditEventsRequest,
 ) {
-  return new Promise<AuditEvents>((resolve, reject) => {
-    if (conn?.facades?.jimM) {
-      conn.facades.jimM
-        .findAuditEvents(params)
-        .then((events) => resolve(events))
-        .catch((error) => reject(error));
-    } else {
-      reject(new Error(Label.NO_JIMM));
-    }
-  });
+  if (!conn.facades.jimM) {
+    throw new Error(Label.NO_JIMM);
+  }
```

Comment:
As you're modifying all these anyway it would probably be a good time to do a little cleanup to create a wrapping function for these that does this check that jimm exists and then throws the error or calls the jimm method, rather than having this repeated in each function.